index.js

```diff
@@ -51,43 +52,57 @@ function filterByKeys(layer, requiredKeys) {
     layer.features = layer.features.filter(function (feature) {
         return hasAllKeys(feature.tags, requiredLookup, requiredCount);
     });
+
+    return layer;
 }
 
-function decorateLayer(layer, keysToKeep, newProps) {
+function selectKeys(layer, keysToKeep) {
     var keys = layer.keys;
     var values = layer.values;
-    var keyLookup = {};
-    var valLookup = {};
+    layer.keyLookup = {};
+    layer.valLookup = {};
     var keepLookup = {};
-
     layer.keys = [];
     layer.values = [];
 
     for (var i = 0; i < keysToKeep.length; i++) {
         keepLookup[keys.indexOf(keysToKeep[i])] = true;
     }
 
-    if (newProps && (newProps.length !== layer.features.length)) {
-        throw new Error('The length of newProps array does not match the number of features');
-    }
-
     for (i = 0; i < layer.features.length; i++) {
         var feature = layer.features[i];
         var tags = feature.tags;
         feature.tags = [];
 
         for (var j = 0; j < tags.length; j += 2) {
             if (keepLookup[tags[j]]) {
-                addKey(keys[tags[j]], keyLookup, layer.keys, feature.tags);
-                addValue(values[tags[j + 1]], valLookup, layer.values, feature.tags);
+                addKey(keys[tags[j]], layer.keyLookup, layer.keys, feature.tags);
+                addValue(values[tags[j + 1]], layer.valLookup, layer.values, feature.tags);
             }
         }
-        if (!newProps) continue;
-        for (var id in newProps[i]) {
-            addKey(id, keyLookup, layer.keys, feature.tags);
-            addValue(newProps[i][id], valLookup, layer.values, feature.tags);
+    }
+
+    return layer;
+}
+
+function decorateLayer(layer, newProps) {
```

Comment:
mmm, yea... I do like `decorate` or `decorateLayer`, but `updateLayerProperties` is more precise